sdk/pinocchio/src/entrypoint/mod.rs

```diff
@@ -89,6 +89,18 @@ pub const SUCCESS: u64 = super::SUCCESS;
 ///
 /// }
 /// ```
+///
+/// # Important
+///
+/// The panic handler set up is different dependending on whether the `std` library is available to the
+/// linker or not. The `entrypoint!` macro will set up a default panic "hook", that works with the
+/// `#[panic_handler]` set by the `std`. Therefore, this macro should be used when the program or any
+/// of its dependencies are dependent on the `std` library.
+///
+/// When the program and all its dependencies are `no_std`, it is necessary to set a
+/// `#[panic_handler]` to handle panics. This is done by the `nostd_panic_handler!` macro. In this case,
+/// it is not possible to use the `entrypoint` macro. Use the `program_entrypoint!` macro instead and set
+/// up the allocator and panic handler manually.
 #[macro_export]
 macro_rules! entrypoint {
```

Comment:
nit: given the information, I almost wonder if we should rename this macro to `std_entrypoint` or `entrypoint_with_std` or `entrypoint_with_allocator_and_panic_handler` to be totally clear

Reply:
Agree, so we could deprecate the `entrypoint` and add an `entrypoint_with_allocator_and_panic_handler` which would do the same.